### PR TITLE
Setting FailureMode from ENV

### PR DESCRIPTION
**Pull request by didierofrivia**
Closes https://github.com/Kuadrant/kuadrant-operator/issues/866

Reading the values from ENV, with the following defaults:

Auth => Deny
RateLimit => Allow

### CODE REVIEW

File: pkg/wasm/types.go
Comment (eguzki) [inline pkg/wasm/types.go:86]: This is breaking kubebuilder validation, isn't it?

Comment (eguzki) [reply pkg/wasm/types.go:86]: Actually, that kubebuilder annotation is useless, isn't it?

Comment (didierofrivia) [reply pkg/wasm/types.go:86]: It was there before, I just added the name for documentation completion on IDEs

Comment (eguzki) [reply pkg/wasm/types.go:86]: I think it is useless...  Anyway, let's not mix doc and annotations

```suggestion
// FailureModeType
+kubebuilder:validation:Enum:=deny;allow
```

File: pkg/wasm/utils.go
Comment (adam-cattermole) [inline pkg/wasm/utils.go:56]: Seems slightly odd to use standard printing here instead of using a logger but I guess that's because it's not easy to get the logger here?

File: tests/envoygateway/extension_reconciler_test.go
Comment (adam-cattermole) [inline tests/envoygateway/extension_reconciler_test.go:174]: I'm sure you've already spotted but the `Timeout` is a pointer whereas the `FailureMode` is not which is why the tests are failing
```suggestion
						FailureMode: wasm.AuthServiceFailureMode(),
```

**Review by adam-cattermole — COMMENTED** [review]
Primarily just tests need fixing but otherwise looks good

File: pkg/wasm/utils.go
Comment (didierofrivia) [reply pkg/wasm/utils.go:56]: Yup, I wanted to avoid having to create an instance of the logger just for this... but I could if needed.

Comment (didierofrivia) [reply pkg/wasm/utils.go:56]: I could also remove the message xD

Comment (adam-cattermole) [reply pkg/wasm/utils.go:56]: I think it's nice to have the message at least to know that if you set it with a typo there's a reason why it's not being set.. but it's also going to look weird with interleaved `Printf` messages with the operator logs every time there's an event that triggers modification to the plugin that goes through this code. I don't want to hold up this PR though not a big issue

**Review by adam-cattermole — APPROVED** [review]
lgtm, verified and all appears to be working

File: pkg/wasm/utils.go
Comment (eguzki) [reply pkg/wasm/utils.go:56]: I would panic, it is wrong to have an invalid value

Comment (eguzki) [reply pkg/wasm/utils.go:56]: But happy to merge this as it is as well

Comment (didierofrivia) [reply pkg/wasm/utils.go]: Take a look at https://github.com/Kuadrant/kuadrant-operator/pull/1030/commits/4bffd5ed4778f30f712b031b006b0c2c5bb6705d, an alternative would be creating a different instance just for the logging that setting or leave it to `fmt.Println`

**Review by eguzki — APPROVED** [review]

**Pull request merged**
